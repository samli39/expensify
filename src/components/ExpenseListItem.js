import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
//parent:ExpenseList

const ExpenseListItem=({id,description,createdAt,amount})=>(
	<div>
		<p>description: <Link to ={`/edit/${id}`}>{description}</Link></p>
		<p>amount: {amount} </p>
		<p>createdAt: { moment(createdAt).format("MMMM Do,YYYY") }</p>
	</div>
)



export default ExpenseListItem;