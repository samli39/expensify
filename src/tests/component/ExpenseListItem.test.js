import React from 'react';
import {shallow} from 'enzyme';
import expenses from '../fixture/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';

test('should render ExpenseListItem correclty',()=>{
	const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
	expect(wrapper).toMatchSnapshot();
})