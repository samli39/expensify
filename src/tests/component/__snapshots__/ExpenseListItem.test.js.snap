// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem correclty 1`] = `
<div>
  <p>
    description: 
    <Link
      replace={false}
      to="/edit/1"
    >
      gum
    </Link>
  </p>
  <p>
    amount: 
    195
     
  </p>
  <p>
    createdAt: 
    December 31st,1969
  </p>
</div>
`;
